osbuild/pipeline.py

```diff
@@ -310,17 +310,13 @@ def assemble(self, object_store, build_tree, tree, monitor, libdir, output_direc
 
         results["assembler"] = r.as_dict()
         if not r.success:
-            output.cleanup()
```

Comment:
This one should be kept, right? The build failed and `output` might have pending cache artifacts, so we should still call `cleanup()`?